### PR TITLE
Feat: Add Support for Integration Actions

### PR DESCRIPTION
**Pull request by korenyoni**
## what
* Add support for the `opsgenie_integration_action` resource in both its own submodule and the `config` submodule.
* update `context.tf` in example configs

## why
* The `opsgenie_integration_action` is useful for manipulating alerts in ways that are not possible with `opsgenie_alert_policy` and `opsgenie_notification_policy`

## references
* https://registry.terraform.io/providers/opsgenie/opsgenie/latest/docs/resources/integration_action

### CODE REVIEW

**Comment by korenyoni** [comment]
```
Error occurred with Status code: 422, Message: Your plan does not allow saving advanced integrations., Took: 0.024000, RequestId: [REDACTED]
```

File: examples/advanced_features/team.tf [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    description    = "team-description"
+    ignore_members = false
+  }
+}
+
+module "sub_team" {
+  source = "../../modules/team"
+
+  team = {
+    name        = "${module.this.id}.team"
```
Comment (aknysh) [inline examples/advanced_features/team.tf:15]: why `.team` ?
maybe give it a more descriptive name?

File: modules/integration_action/variables.tf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+variable "integration_action" {
+  default     = {}
+  description = "Opsgenie API Integration configuration"
```
Comment (aknysh) [inline modules/integration_action/variables.tf:3]: is this correct description?

Comment (aknysh) [reply modules/integration_action/variables.tf:3]:
```
variable "integration_action" {  default     = {}  description = "Opsgenie Integration Action configuration"}
```

**Review by aknysh — COMMENTED** [review]
LGTM, please see comments

**Review by aknysh — APPROVED** [review]

**Pull request merged**